src/main/java/hudson/plugins/sauce_ondemand/credentials/SauceCredentials.java

```diff
@@ -111,20 +111,12 @@ public String getDisplayName() {
 
         @SuppressWarnings("unused") // used by stapler
         public FormValidation doCheckApiKey(@QueryParameter String value, @QueryParameter String username) {
-            return doValidate(username, value);
-        }
-
-        @SuppressWarnings("unused") // used by stapler
-        public FormValidation doValidate(@QueryParameter String username, @QueryParameter String apiKey) {
-            if (actualValidate(username, apiKey))
+            SauceREST rest = new JenkinsSauceREST(username, value);
+            if (!rest.getUser().equals("")) {
```

Comment:
I agree, i was thinking to refactor some more to actually check 401 unauthorized, as that would be the actual correct way to deal with this.